uvloop/dns.pyx

```diff
@@ -382,7 +377,13 @@ cdef class AddrInfoRequest(UVRequest):
 
         if err < 0:
             self.on_done()
-            callback(convert_error(err))
+            if err == uv.UV_EINVAL:
+                # Convert UV_EINVAL to EAI_NONAME to match libc behavior
+                msg = system.gai_strerror(socket_EAI_NONAME).decode('utf-8')
```

Comment:
Can this line fail with an exception?

Reply:
I suppose it can, though `convert_error()` would have failed too; but I'll add an error handling.